### PR TITLE
Jetpack: Integrate the new disconnect dialog block

### PR DESCRIPTION
**Pull request by enejb**
Lets Integrate the new Jetpack Disconnect Dialog into the Jetpack Disconnect Button

To test:
Go to Settings General and click the disconnect button. 
Do you see any error in console when disconnecting the site?

### CODE REVIEW

File: client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx [hunk 1]
```diff
@@ -3,36 +3,95 @@
...
+			// `receiveDeletedSiteDeprecated` here because the site
+			// exists in `sites-list` as well as the global store.
+			disconnectedSiteDeprecated( site );
+			this.props.setAllSitesSelected();
+			removeInfoNotice( notice.noticeId );
+			showSuccessNotice( translate( 'Disconnected %(siteName)s', { args: { siteName: site.title } } ) );
+			recordGAEvent( 'Jetpack', 'Successfully Disconnected' );
+		}, () => {
+			removeInfoNotice( notice.noticeId );
+			showErrorNotice( translate( 'Error Disconnecting %(siteName)s', { args: { siteName: site.title } } ) );
```
Comment (a8ci18n) [inline client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Error disconnecting %(siteName)s.' )` *ES Score: 18*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/disable-jetpack-dialog)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx [hunk 2]
```diff
@@ -3,36 +3,95 @@
...
+
+		disconnectSite( site.ID ).then( () => {
+			// Removing the domain from a domain-only site results
+			// in the site being deleted entirely. We need to call
+			// `receiveDeletedSiteDeprecated` here because the site
+			// exists in `sites-list` as well as the global store.
+			disconnectedSiteDeprecated( site );
+			this.props.setAllSitesSelected();
+			removeInfoNotice( notice.noticeId );
+			showSuccessNotice( translate( 'Disconnected %(siteName)s', { args: { siteName: site.title } } ) );
```
Comment (a8ci18n) [inline client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Successfully disconnected %(siteName)s.' )` *ES Score: 18*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/disable-jetpack-dialog)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx [hunk 3]
```diff
@@ -3,36 +3,95 @@
...
+			removeNotice: removeInfoNotice,
+			disconnect: disconnectSite,
+			recordGoogleEvent: recordGAEvent
+		} = this.props;
+
+		this.setState( { dialogVisible: false } );
+		recordGAEvent( 'Jetpack', 'Clicked To Confirm Disconnect Jetpack Dialog' );
+
+		const { notice } = showInfoNotice(
+			translate( 'Disconecting %(siteName)s', { args: { siteName: site.title } } )
```
Comment (a8ci18n) [inline client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Disconnecting %(siteName)s.' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/disable-jetpack-dialog)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by lezama** [comment]
I am getting this error right after disconnecting

```warning.js:36 Warning: Failed prop type: Invalid prop `site` of type `boolean` supplied to `SiteSettingsNavigation`, expected `object`.
    in SiteSettingsNavigation (created by SiteSettingsComponent)
    in SiteSettingsComponent (created by Connect(SiteSettingsComponent))
    in Connect(SiteSettingsComponent)
    in Provider```

**Comment by enejb** [comment]
@lezama Just fixed the JS error.

**Comment by rickybanister** [comment]
This worked great. 

My only feedback would be a small tweak to the successful disconnection notice:

Let's change it from `Disconnected Site Name` to `Site Name has been successfully disconnected`

**Comment by rickybanister** [comment]
One other bit, I tried reconnecting via the 'add self-hosted site' sidebar menu right afterward. It worked fine, but the initial 'connecting' screen made me wait more than 30 seconds before it showed me the 'Connecting as...' screen, which also took a very long time before I saw an 'Approve' button. 

The connect flow is normally quite quick for me. Is there anything about reconnecting immediately that would slow this down—is this due to refreshing the sites list and the number of sites I have?

**Comment by lezama** [comment]
> The connect flow is normally quite quick for me. Is there anything about reconnecting immediately that would slow this down—is this due to refreshing the sites list and the number of sites I have?

In any case it shouldn't be related to this PR.

Code looks good to me 👍, let's merge it :)

File: client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx [hunk 4]
```diff
@@ -3,36 +3,95 @@
...
+
+		disconnectSite( site.ID ).then( () => {
+			// Removing the domain from a domain-only site results
+			// in the site being deleted entirely. We need to call
+			// `receiveDeletedSiteDeprecated` here because the site
+			// exists in `sites-list` as well as the global store.
+			disconnectedSiteDeprecated( site );
+			this.props.setAllSitesSelected();
+			removeInfoNotice( notice.noticeId );
+			showSuccessNotice( translate( '%(siteName)s has been successfully disconnected', { args: { siteName: site.title } } ) );
```
Comment (a8ci18n) [inline client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Successfully disconnected %(siteName)s.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx [hunk 5]
```diff
@@ -3,36 +3,95 @@
...
+			removeNotice: removeInfoNotice,
+			disconnect: disconnectSite,
+			recordGoogleEvent: recordGAEvent
+		} = this.props;
+
+		this.setState( { dialogVisible: false } );
+		recordGAEvent( 'Jetpack', 'Clicked To Confirm Disconnect Jetpack Dialog' );
+
+		const { notice } = showInfoNotice(
+			translate( 'Disconnecting %(siteName)s', { args: { siteName: site.title } } )
```
Comment (a8ci18n) [inline client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Disconnecting %(siteName)s.' )` *ES Score: 25*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/disable-jetpack-dialog)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**